### PR TITLE
[BISERVER-14305] - Admin doesn't see deleted file in the trash folder

### PR DESCRIPTION
**Pull request by bantonio82**
This PR is to fix the corner case when a user home folder is deleted.

@pentaho-lmartins  @ssamora

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 26m 56s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl repository
```
---
:ok_hand: All tests passed!
 
Tests run: **1866**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4628/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: repository/src/main/java/org/pentaho/platform/repository2/unified/jcr/DefaultDeleteHelper.java [hunk 1]
```diff
@@ -370,11 +371,16 @@ protected boolean isAdmin() {
...
-          deletedFiles.addAll( deletedForUser );
+        try {
+          Node userHomeFolderNode = (Node) session.getItem( homePath );
+          if ( userHomeFolderNode.hasNode( FOLDER_NAME_TRASH ) ) {
+            Node trashNode = userHomeFolderNode.getNode( FOLDER_NAME_TRASH );
+            List<RepositoryFile> deletedForUser = getDeletedFiles( session, pentahoJcrConstants, trashNode, user );
+            deletedFiles.addAll( deletedForUser );
+          }
+        } catch ( PathNotFoundException e ) {
+          System.out.println( "hello" );
```
Comment (pentaho-lmartins) [inline repository/src/main/java/org/pentaho/platform/repository2/unified/jcr/DefaultDeleteHelper.java]: What's this?

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 26m 5s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl repository
```
---
:ok_hand: All tests passed!
 
Tests run: **1866**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4628/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 26m 6s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl repository
```
---
:ok_hand: All tests passed!
 
Tests run: **1866**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4628/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**